app/models/concerns/hostext/search.rb

```diff
@@ -20,6 +20,8 @@ module Search
       scoped_search :on => :owner_type,    :complete_value => true, :only_explicit => true
       scoped_search :on => :owner_id,      :complete_enabled => false, :only_explicit => true, :validator => ScopedSearch::Validators::INTEGER
       scoped_search :on => :id,            :complete_enabled => false, :only_explicit => true, :validator => ScopedSearch::Validators::INTEGER
+      loader_values = Operatingsystem.all_loaders_map.keys
+      scoped_search :on => :pxe_loader, :complete_value => loader_values.map { |v| [v.tr(' ', '_').to_sym, v] }.to_h, :only_explicit => true, :operators => ['=']
```

Comment:
Does assignment to an additional variable assignment here mean it gets stored in each class that inherits this?

Reading it again, you can even use `.to_h` in block form to further remove method calls:
```suggestion
      scoped_search :on => :pxe_loader, :complete_value => Operatingsystem.all_loaders_map.to_h { |k, _v| [k.tr(' ', '_').to_sym, k] }, :only_explicit => true, :operators => ['=']
```

Reply:
:+1: yeah, accepting not because of the memory saving,which I think is minimal, but because it's much more readable